flake.nix

```
@@ -17,22 +20,67 @@
     };
   };
 
-  outputs = { self, nixpkgs, flake-utils, cardano-node }:
+  outputs = { self, nixpkgs, flake-utils, cardano-node, poetry2nix }:
     flake-utils.lib.eachDefaultSystem
       (system:
         let
           pkgs = nixpkgs.legacyPackages.${system};
-          python3 = pkgs.python3.override {
-            packageOverrides = self: _: {
-              allure = self.callPackage ./nix/allure.nix { };
-              pytest-allure = self.callPackage ./nix/pytest-allure.nix { };
-              pytest-select = self.callPackage ./nix/pytest-select.nix {};
-              cardano-clusterlib = self.callPackage ./nix/cardano-clusterlib.nix { };
-            };
+          p2n = poetry2nix.legacyPackages.${system};
+          # Workarounds `ModuleNotFoundError: No module named 'poetry'`
+          # cf. https://github.com/nix-community/poetry2nix/blob/master/docs/edgecases.md#modulenotfounderror-no-module-named-packagename
+          pypkgs-build-requirements = {
+            pytest-metadata = [ "poetry" ];
+            pytest-select = [ "poetry" ];
+            m2r2 = [ "setuptools" ];
+            sphinxemoji = [ "setuptools" ];
+            sphinxcontrib-jquery = [ "flit-core" ];
+          };
+          p2n-overrides = p2n.defaultPoetryOverrides.extend (self: super:
+            builtins.mapAttrs
+              (package: build-requirements:
+                (builtins.getAttr package super).overridePythonAttrs (old: {
+                  buildInputs = (old.buildInputs or [ ]) ++ (builtins.map (pkg: if builtins.isString pkg then builtins.getAttr pkg super else pkg) build-requirements);
```

Could this be improved by selectively using `with builtins;`?
```suggestion
                  buildInputs = (old.buildInputs or [ ])
                             ++ (with builtins; map (pkg: if isString pkg then getAttr pkg super else pkg) build-requirements);
```
isn't `getAttr pkg super` identical to `super.${pkg}`? Which _I_ find more readable, maybe I'm an outlier.
So this would be `(with builtins; map (pkg: if isString pkg then super.${pkg} else pkg)`.